### ISSUE TITLE
Auto-port detection code not working as intended

### ISSUE THREAD

**Issue by Ownasaurus** [Ownasaurus/BK-390A#1]
Nullkraft commented
OK, I see that you've already incorporated Paul's changes. I tested for the flashing 'm'. Anyway, I ran it without any command line parameters and it of course took 2 seconds to find the port but it did find it. Next I ran it with -d and finally I ran it with -p 3 -d and everything seems to be working as advertised.

I noticed a faster startup when specifying the port number. You might want to add an artificial delay so the end user doesn't think something is wrong with auto detection. Just saying. ;)

pvtinflex commented
Maybe make it write up "Checking port... ", ah but all the comms detection happens before the GDI starts, though you could swap that around.

**Comment by Ownasaurus** [comment]
@Nullkraft what was the debug output when it failed to detect the port? That output would be helpful. Thanks!

**Comment by Ownasaurus** [comment]
First I will work on getting it functional, and then I will make it more pretty. Showing something like "Searching for device...." in the GUI would be excellent.

**Comment by Nullkraft** [comment]
Just checking in real quick.  I think there's some confusion.  The port detection worked.  The only thing I mentioned that users might complain about is that **manually** choosing a port starts the program faster.

**Comment by Ownasaurus** [comment]
My mistake! I thought it found the port but it didn't work anyway, and that's why you tried -d. My flawed assumption. Sweet! In that case I'll definitely work on adding a GUI notification and/or decreasing the sleep time.  Thanks for clarification.

**Comment by Ownasaurus** [comment]
@Nullkraft Can you please see if the auto detection code is still functional now? If so it should be much improved.

**Comment by Nullkraft** [comment]
Alright, rebooting to windows.  I'll let you know in a couple of minutes.

**Comment by Nullkraft** [comment]
I'm not sure what is going on.  I am using an USB-to-serial adapter and when I rebooted to start windows the win-bk390a.exe failed to open the com port.

C:\Users\mark\Downloads\bk390_app>win-bk390a.exe
Port COM3 Opened
        Baudrate = 2400
        ByteSize = 7
        StopBits = 0
        Parity   = 1
        Setting time-outs successful
        CommMask successful
Error while trying to open com port 'COM3'

I tried twice and then I removed and reinserted the usb adapter and everything works fine.  I'm going to try it again and see what happens.  Give me a couple of minutes.

**Comment by Ownasaurus** [comment]
It could be an error in my adjusted algorithm too, so don't knock your hardware yet!
I probably won't be able to edit any more code until tomorrow though.

**Comment by Nullkraft** [comment]
No problem.  I've eliminated my hardware as the problem.  It seems that the timing of closing/opening the port **may** be the problem.  If I try to open immediately after exiting the app I randomly get the error.  Also, once I get the error if I repeatedly, and quickly, attempt to reconnect I'll get the error several times in a row.  But not always.

**Comment by Nullkraft** [comment]
I'll ping you when I get home from work tomorrow.

**Comment by Ownasaurus** [comment]
Increased timeout period from 500ms to 1000ms. Let's see if that fixes the issue.

**Comment by Nullkraft** [comment]
Sorry I'm late, I was watching Louis.  Starting test now.

**Comment by Ownasaurus** [comment]
OK. I'm available for the next hour or so. Do you want to do a gchat or zoom meeting or something to facilitate rapid testing?

**Comment by Nullkraft** [comment]
No clue.  Which one let's you see what's happening and how do I get it up and going?

**Comment by Ownasaurus** [comment]
Both. If you give me your email address, we can make it happen quickly and easily

**Comment by Ownasaurus** [comment]
i dont know how to PM on github though

**Comment by Nullkraft** [comment]
So no PM then...  nullkraft@gmail.com for google but if it will work I'd rather mstanley@technologist.com

**Comment by Ownasaurus** [comment]
Message sent

**Comment by Nullkraft** [comment]
Thankyou!

**Comment by Ownasaurus** [comment]
Did you receive the email?

**Comment by Nullkraft** [comment]
Got it but I don't use, or have, a microphone.

**Comment by Ownasaurus** [comment]
Thanks for everything. I think we resolved a lot today :)

**Comment by Nullkraft** [comment]
I should have put this over here -

"I just came from watching a couple more of his videos and it is clear to me that he set the meter to a fixed range. When the voltage drops below about 0.470 V it would normally autorange into the milliVolt region. But it never switches even when you see a number like 0.158 V. Fixed Range...."

**Comment by Ownasaurus** [comment]
No worries. I don't think he ever updated the software yet.

**Comment by Nullkraft** [comment]
I posted this on inflex github also:

I tried the command line: win-bk390a.exe -z 120 -p 3 -m -fc #ff1010 -bc #000000 -fw 600
and everything seems to be working fine. I changed -z a few times and had good control. I got a rather large red font in the display.

I will mention one thing, when I set -z to a relatively small value (less than 25) the display began to cutoff portions of the information, namely the description(?) such as the word 'Frequency'.
